### PR TITLE
 fetch data

### PR DESCRIPTION
**Pull request by developer-digital**
fetch data from api youtube 
https://edu-redmine.framgia.vn/issues/6677

### CODE REVIEW

**Comment by developer-digital** [comment]
ready

File: app/src/main/java/com/framgia/youtube_background_01/data/source/VideoDataSource.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+    interface RemoteDataSource {
+        void getTrendingVideos(OnFetchDataListener<Video> listener);
+
+        void getVideosByPlaylist(String genreType, OnFetchDataListener<Video> listener);
+
+        void querySearch(String keyword, OnFetchDataListener<Video> listener);
+    }
+}
+
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/framgia/youtube_background_01/data/source/VideoDataSource.java]: remove blank

File: app/src/main/java/com/framgia/youtube_background_01/data/source/remote/VideoRemoteDataSource.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    @Override
+    public void getVideosByPlaylist(String idList, OnFetchDataListener<Video> listener) {
+        new GetVideosAsyncTask(QueryType.GET_VIDEOS, listener).execute(
+                BASE_URL + BASE_PLAYLIST + idList + BASE_MAX_RESULTS + BASE_API_KEY);
+    }
+
+    @Override
+    public void querySearch(String keyword, OnFetchDataListener<Video> listener) {
+        new GetVideosAsyncTask(QueryType.GET_VIDEOS, listener).execute(
+                BASE_URL + BASE_SEARCH + keyword + BASE_MAX_RESULTS + BASE_API_KEY);
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/framgia/youtube_background_01/data/source/remote/VideoRemoteDataSource.java]: BASE_URL, BASE_SEARCH, BASE_MAX_RESULTS, BASE_API_KEY
những trường này lặp nhau. refactor code đi e

**Comment by developer-digital** [comment]
ready

File: app/src/main/java/com/framgia/youtube_background_01/utils/Constant.java [hunk 3]
```diff
@@ -9,10 +9,12 @@
...
     public static final String THUMBNAILS = "thumbnails";
     public final static String BASE_URL = "https://www.googleapis.com/youtube/v3/";
-    public final static String BASE_PLAYLIST = "playlistItems?part=snippet&playlistId=";
-    public final static String BASE_SEARCH = "search?part=snippet&q=";
-    public final static String BASE_URL_IMAGE = "https://i.ytimg.com/vi/";
+    public final static String BASE_QUERY_PLAYLIST = "playlistItems?part=snippet&playlistId=";
     public final static String BASE_MAX_RESULTS = "&maxResults=50";
     public final static String BASE_TRENDING = "PLUadgMpPaifVmhXn4xz-jRO934EAORUnX";
-    public final static String BASE_API_KEY = "&key=" + BuildConfig.API_KEY;
+    public final static String BASE_PLAYLIST_URL =BASE_URL+BASE_QUERY_PLAYLIST;
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/framgia/youtube_background_01/utils/Constant.java]: reformat code

**Comment by developer-digital** [comment]
ready

**Review by tuanlvt-1117 — APPROVED** [review]

**Pull request merged**